aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -1206,14 +1206,37 @@ Tensor std_mps(const Tensor& input_t,
   @autoreleasepool {
     string key = string("any_out_mps:") + getTensorsStringKey(input_t) + ":" + std::to_string(dim_);
     auto cachedGraph = LookUpOrCreateCachedGraph<CachedGraph>(key, [&](auto mpsGraph, auto newCachedGraph) {
-      MPSGraphTensor* inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, input_t);
+      auto inputTensor = mpsGraphRankedPlaceHolder(mpsGraph, input_t);
 
-      MPSGraphTensor* castInputTensor =
-          castToIHFTypes(mpsGraph, inputTensor, input_t, /*includesInt64=*/macOS13_3_plus);
-      MPSGraphTensor* castOutputTensor = [mpsGraph reductionOrWithTensor:castInputTensor axis:dim_ name:nil];
-      MPSGraphTensor* outputTensor = castOutputTensor;
-      if (MPSDataTypeBool != [castOutputTensor dataType]) {
-        outputTensor = [mpsGraph castTensor:castOutputTensor toType:MPSDataTypeBool name:@"outputTensor"];
+      auto castInputTensor = castToIHFTypes(mpsGraph, inputTensor, input_t, /*includesInt64=*/macOS13_3_plus);
+      // reductionOrWithTensor:axis: will throw an internal assert if number of dimentions is more than 4
+      // See https://github.com/pytorch/pytorch/issues/95538
+      MPSGraphTensor* outputTensor = nil;
+      if (input_t.ndimension() > 4) {
+        auto reduceDimLen = input_t.size(dim_);
+        if (dim_ == 0) {
+          castInputTensor = [mpsGraph reshapeTensor:castInputTensor withShape:@[ @(reduceDimLen), @-1 ] name:nil];
```

Comment:
I guess you can always do this reshape because the Tensor here is always contiguous?

Reply:
To be frank, I don't know how MPS does reshapes under the hood, but currently there is no way to define a non-contiguous MPS tensor.